stdlib/importlib/metadata/__init__.pyi

```diff
@@ -109,8 +134,14 @@ if sys.version_info >= (3, 8):
     def distributions(
         *, context: None = ..., name: str | None = ..., path: list[str] = ..., **kwargs: Any
     ) -> Iterable[Distribution]: ...
-    def metadata(distribution_name: str) -> Message: ...
+
+    if sys.version_info >= (3, 10):
+        def metadata(distribution_name: str) -> PackageMetadata: ...
+        def entry_points() -> SelectableGroups: ...
+    else:
+        def metadata(distribution_name: str) -> Message: ...
+        def entry_points() -> dict[str, list[EntryPoint]]: ...
```

Comment:
I think this is incomplete, consider the example:
```py
(mypy,) = entry_points(group="console_scripts", name="mypy")
print(mypy)
```
This is valid and works:
```
EntryPoint(name='mypy', value='mypy.__main__:console_entry', group='console_scripts')
```

![image](https://user-images.githubusercontent.com/12860284/154865432-75101f23-b095-4c16-9df3-71a102a0aaf7.png)

Reply:
This is why I chose to use `list`:

https://github.com/python/cpython/blob/b77158b4da449ec5b8f682816a79d004fd65ed07/Lib/importlib/metadata/__init__.py#L325

Using `tuple` instead doesn't make much sense to me, as it would contradict the CPython implementation.

Reply:
My point was about 3.10 if branch, sorry for commenting on the wrong line.

https://github.com/python/typeshed/blob/a10cced8ea144af367f70349c65f475cd81707aa/stdlib/importlib/metadata/__init__.pyi#L140

This function should have `name` and `groups` arguments.

Reply:
Updated, see the discussion below.

Reply:
Now I get this:
```
stdlib\importlib\metadata\__init__.pyi:141: error: Overloaded function signatures 1 and 2 overlap with incompatible return types
```

Reply:
Error is now ignored.